WeightProducer/python/MCSampleValues.py

```diff
@@ -63,493 +63,1039 @@ class MCSampleValuesHelper():
+            ),
         },
         "WJetsToLNu_HT-800To1200" : {
-            "CrossSection" : XSValues(XS_13TeV=6.66),
-            "kFactor" : kFactorValues(kFactor_13TeV=1.21),
+            "CrossSection" : XSValues(
+                XS_2016=5.497, XSSource_2016="XSDB (LO)",
+                XS_2017=5.366, XSSource_2017="XSDB (LO)",
+                XS_2018=5.451, XSSource_2018="GenXSecAnalyzer",
+            ),
+            "kFactor" : kFactorValues(
+                kFactor_2016=1.224, kFactorSource_2016="XSDB NNLO/LO=61526.7/50260",
+                kFactor_2017=1.162, kFactorSource_2017="XSDB NNLO/LO=61526.7/52940",
+                kFactor_2018=1.164, kFactorSource_2018="XSDB NNLO/LO=61526.7/52850",
+            ),
         },
         "WJetsToLNu_HT-1200To2500" : {
-            "CrossSection" : XSValues(XS_13TeV=1.608),
-            "kFactor" : kFactorValues(kFactor_13TeV=1.21),
+            "CrossSection" : XSValues(
+                XS_2016=1.329, XSSource_2016="XSDB (LO)",
+                XS_2017=1.074, XSSource_2017="XSDB (LO)",
+                XS_2018=1.085, XSSource_2018="GenXSecAnalyzer",
+            ),
+            "kFactor" : kFactorValues(
+                kFactor_2016=1.224, kFactorSource_2016="XSDB NNLO/LO=61526.7/50260",
+                kFactor_2017=1.162, kFactorSource_2017="XSDB NNLO/LO=61526.7/52940",
+                kFactor_2018=1.164, kFactorSource_2018="XSDB NNLO/LO=61526.7/52850",
+            ),
         },
         "WJetsToLNu_HT-2500ToInf" : {
-            "CrossSection" : XSValues(XS_13TeV=0.03891),
-            "kFactor" : kFactorValues(kFactor_13TeV=1.21),
+            "CrossSection" : XSValues(
+                XS_2016=0.03209, XSSource_2016="XSDB (LO)",
+                XS_2017=0.008001, XSSource_2017="XSDB (LO)",
+                XS_2018=0.008060, XSSource_2018="GenXSecAnalyzer",
+            ),
+            "kFactor" : kFactorValues(
+                kFactor_2016=1.224, kFactorSource_2016="XSDB NNLO/LO=61526.7/50260",
+                kFactor_2017=1.162, kFactorSource_2017="XSDB NNLO/LO=61526.7/52940",
+                kFactor_2018=1.164, kFactorSource_2018="XSDB NNLO/LO=61526.7/52850",
+            ),
         },
         "WJetsToLNu" : {
-            "CrossSection" : XSValues(XS_13TeV=61334.9),
-        },
-        "WJetsToQQ_HT400to600_qc19_3j" : {
-            "CrossSection" : XSValues(XS_13TeV=316.4),
-        },
-        "WJetsToQQ_HT600to800_qc19_3j" : {
-            "CrossSection" : XSValues(XS_13TeV=68.54),
-        },
-        "WJetsToQQ_HT-800toInf_qc19_3j" : {
-            "CrossSection" : XSValues(XS_13TeV=34.68),
-        },
-        "WJetsToQQ_HT-600ToInf" : {
-            "CrossSection" : XSValues(XS_13TeV=95.14),
-            "kFactor" : kFactorValues(kFactor_13TeV=1.21),
+            "CrossSection" : XSValues(
+                XS_13TeV=61526.7, XSSource_13TeV="https://twiki.cern.ch/twiki/bin/viewauth/CMS/SummaryTable1G25ns#W_jets NNLO (60430.0 @ NLO)",
```

Comment:
is there a reason to keep this one? should a k-factor be assessed?

Reply:
I used these values for assessing the k-factors used in the HT binned samples above. I don't think it hurts to have this value there given that the code should default to the 13TeV value and only go to the year-by-year values if it doesn't exist. I'm double checking that now.